enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,435 @@
+---
+title: platform-operators
+authors:
+  - "@timflannagan"
+reviewers:
+  - TBD
+approvers:
+  - "@joelanford"
+  - "@bparees"
+  - TBD
+api-approvers:
+  - TBD
+creation-date: 2022-06-30
+last-updated: 2022-07-06
+tracking-link:
+  - https://issues.redhat.com/browse/OLM-2232
+  - https://issues.redhat.com/browse/OLM-2513
+  - https://issues.redhat.com/browse/OLM-2605
+see-also:
+  - https://issues.redhat.com/browse/OLM-2447
+  - https://issues.redhat.com/browse/OLM-2570
+  - https://issues.redhat.com/browse/CORS-1873
+  - https://issues.redhat.com/browse/OCPPLAN-7589
+  - https://github.com/openshift/enhancements/blob/master/enhancements/update/cvo-techpreview-manifests.md
+replaces: []
+superseded-by: []
+---
+
+# Platform Operators
+
+## Summary
+
+The enhancement outlines a mechanism that enables support for the lifecycle of platform operators. A platform operator (PO) is an OLM-based cluster extension that can be installed during day 0 operations, and lifecycles with the cluster. This mechanism will ultimately be divided into phases, and this proposal covers the first phase in more detail.
+
+This first phase introduces a basic PO installation mechanism, and helps build the foundational layer for subsequent phases. Future phases may introduce a first-class PO management mechanism for the OLM component.
+
+## Motivation
+
+The motivation for platform operators is two-fold. The strategic direction for OpenShift is to make the core installation smaller by default and at the same time make it more flexible so that customers can tailor their installations to their use cases. This is what we’re calling “Composable OpenShift”.
+
+To make the core smaller, OpenShift is attempting to reduce the CVO-managed payload to the very minimum set of operators required to run the OpenShift control plane. From a control plane perspective, any non-essential operators can be installed after the cluster has been installed (or upgraded). However, from a customer standpoint, their business needs may dictate that these non-essential operators are still included in the lifecycle of cluster installations and upgrades. The Platform Operator (PO) concept will allow cluster administrators to choose which of these non-control-plane-essential operators to include in the cluster lifecycle.
```

Comment:
i think we are talking about "non-essential" operators as those that aren't necessary for the control plane to function, but i am curious about possible use cases where a platform integrator would want to supply a PO for a low level component like the Cloud Controller Manager. we currently have an operator that manages the deployment of CCMs, but we are looking at ways to allow integrators to replace that component so that they can provide platform specific components for their installations.  i'm curious if you have any thoughts or comments about this non-essential definition and how it might apply to things like a CCM operator?

Reply:
yeah POs can really only cover components that the control plane and OLM itself do not depend on.  If we can get to a running control plane + olm w/o it, then it should be viable as a PO.

Reply:
We have a couple of different threads on the potential use cases around MAPI/CCM. I think I'm incline to make that an explicit non-goal for this proposal, but I'm still trying to figure out what's the best way to track this potential use case so we re-evaluate during subsequent phase designs. Thoughts?

Reply:
is there some way we can capture the future intent to extend the platform operators to cover things like the MAPI and CCM operators?

i have a feeling this will extend to more than just the MAPI and CCM operators, there are other platform-specific components that our users would like the ability to swap.

Reply:
Updated in the last commit (https://github.com/openshift/enhancements/pull/1170/commits/845b885d84cf03f943a469c6fcf859e4bb74f33d). I added some language in the non-goals, goals, open questions, etc.. We don't have a dedicated phase 1 downstream Epic yet, but I can add a spike ticket so we're covered on all fronts.

Reply:
thanks @timflannagan i like that we are calling it out specifically

Reply:
Yep, no problem. Marking this thread as resolved now :+1:.